sensor/kubernetes/listener/resources/networkpolicy_test.go

```diff
@@ -0,0 +1,445 @@
+package resources
+
+import (
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stackrox/rox/generated/internalapi/central"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/features"
+	"github.com/stackrox/rox/sensor/common/detector/mocks"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+func TestGetSelector(t *testing.T) {
+	if !features.NetworkPolicySystemPolicy.Enabled() {
+		return
+	}
+
+	mockCtrl := gomock.NewController(t)
+	nps := newNetworkPoliciesStore()
+	ds := newDeploymentStore()
+	det := mocks.NewMockDetector(mockCtrl)
+
+	dispatcher := newNetworkPolicyDispatcher(nps, ds, det)
+
+	cases := []struct {
+		name             string
+		netpol           *storage.NetworkPolicy
+		oldNetpol        *storage.NetworkPolicy
+		action           central.ResourceAction
+		expectedSelector []map[string]string
+		expectedEmpty    bool
+	}{
+		{
+			name: "New NetworkPolicy",
+			netpol: &storage.NetworkPolicy{
```

Comment:
It seems like `Id` and `Namespace` are the same for all the tests cases.

1) Does it make sense to test different values?
2) If not, a nitpick: maybe you can create a function to wrap the creation of test Network Policies? This way all the constant values that are present in all policies won't need to show up in every test case. It makes it easier to read also since the test case will only hold necessary information for this assertion.

Reply:
In some tests the `namespace` is different but I still think that your suggestion of wrapping the creation of `storage.NetworkPolicy`s is valid.